src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs

```diff
@@ -181,6 +183,11 @@ private MessageType GetMessageType(ReadOnlySpan<byte> line)
         {
             EnsureStartsWithDataPrefix(line);
 
+            if (line.Length != _messagTypeLineLength)
```

Comment:
Maybe just rename this function to `ParseMessageType` then? We are already ensuring that the line starts with "data: " so it is more just a simple conversion function. I am not a big fan of `"data: X\r\n"` but it's probably fine. `_messagTypeLineLength` has a typo.